kola/harness.go

```diff
@@ -418,16 +418,30 @@ func RunTest(t *register.Test, pltfrm string) error {
 		}
 	}
 
-	// run test
-	err = t.Run(tcluster)
-
 	// give some time for the remote journal to be flushed so it can be read
 	// before we run the deferred machine destruction
-	if err != nil {
-		time.Sleep(10 * time.Second)
-	}
+	defer func() {
+		if err != nil {
+			time.Sleep(2 * time.Second)
+		}
+	}()
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch r := r.(type) {
+			// can't have multiple types in a single case.
+			case error:
+				err = r
+			case skip.Skip:
```

Comment:
interesting point, but according to my test, the concrete type is chosen over the interface, although i'm unsure if this is documented somewhere...

```
$ sudo ./bin/kola --qemu-image ~/src/coreos/src/build/images/amd64-usr/latest/coreos_production_image.bin run coreos.verity.corruption   
2016-05-17T23:44:35Z kola: === RUN coreos.verity.corruption on qemu
2016-05-17T23:44:47Z kola: --- SKIP: coreos.verity.corruption on qemu (11.985s)
2016-05-17T23:44:47Z kola:         verity is not enabled
2016-05-17T23:44:47Z kola: 0 passed 0 failed 1 skipped out of 1 total
```